crates/project/src/worktree_store.rs

```diff
@@ -302,7 +314,12 @@ impl WorktreeStore {
     }
 
     pub fn add(&mut self, worktree: &Model<Worktree>, cx: &mut ModelContext<Self>) {
-        let push_strong_handle = self.is_shared || worktree.read(cx).is_visible();
+        let worktree_id = worktree.read(cx).id();
```

Comment:
Let's make this an assert and add `#[track_caller]` to the function, so we could see what caused the duplication to occur